cmd/private-prow-configs-mirror/main.go

```diff
@@ -290,73 +290,55 @@ func injectPrivateBugzillaPlugin(bugzillaPlugins plugins.Bugzilla, orgRepos orgR
-			org := strings.Split(orgRepo, "/")[0]
-			if _, ok := plugins[org]; ok {
-				newValue = append(newValue, plugins[org]...)
+			if repoValues, ok := plugins[fmt.Sprintf("%s/%s", org, repo)]; ok {
+				values.Insert(repoValues...)
 			}
-
-			privateReposWithValues[privateOrgRepo(strings.Split(orgRepo, "/")[1])] = newValue
-			allPrivate = append(allPrivate, newValue...)
+			privateReposWithValues[privateOrgRepo(repo)] = values.List()
 		}
 	}
 
-	mergeCommonPrivatePlugins(plugins, privateReposWithValues, getRepeatedValues(allPrivate))
-}
-
-// mergeCommonPrivatePlugins detects the common values from all the private repositories
-// and merges them into the org `openshift-priv` level.
-// In addition, it generates a repo. if it contains an extra non-common field.
-func mergeCommonPrivatePlugins(plugins map[string][]string, privateReposWithValues map[string][]string, repeatedValues sets.String) {
-	privateOrgPlugins := sets.NewString()
-
-	for repoName, repoValues := range privateReposWithValues {
-		repoLevelPlugins := sets.NewString()
-		valuesSet := sets.NewString(repoValues...)
+	commonValuesSet := getCommonValues(privateReposWithValues)
+	for repo, values := range privateReposWithValues {
+		repoLevelPlugins := sets.NewString(values...)
 
-		if len(repeatedValues) > 0 {
-			privateOrgPlugins = privateOrgPlugins.Union(valuesSet.Intersection(repeatedValues))
-			repoLevelPlugins = repoLevelPlugins.Union(valuesSet.Difference(repeatedValues))
-		} else {
-			repoLevelPlugins.Insert(repoValues...)
-		}
+		repoLevelPlugins = repoLevelPlugins.Difference(commonValuesSet)
 
 		if len(repoLevelPlugins.List()) > 0 {
-			logrus.WithFields(logrus.Fields{"repo": repoName, "value": repoLevelPlugins.List()}).Info("Generating repo")
-			plugins[repoName] = repoLevelPlugins.List()
+			logrus.WithFields(logrus.Fields{"repo": repo, "value": repoLevelPlugins.List()}).Info("Generating repo")
+			plugins[repo] = repoLevelPlugins.List()
 		}
 	}
 
-	if len(privateOrgPlugins.List()) > 0 {
-		logrus.WithField("value", privateOrgPlugins.List()).Info("Generating openshift-priv org.")
-		plugins[openshiftPrivOrg] = privateOrgPlugins.List()
+	if len(commonValuesSet.List()) > 0 {
+		logrus.WithField("value", commonValuesSet.List()).Info("Generating openshift-priv org.")
+		plugins[openshiftPrivOrg] = commonValuesSet.List()
 	}
 }
 
-func getRepeatedValues(values []string) sets.String {
-	temp := sets.NewString()
-	repeated := sets.NewString()
+func getCommonValues(privateReposWithValues map[string][]string) sets.String {
+	ret := sets.NewString()
+	count := 0
+	for _, values := range privateReposWithValues {
+		valuesSet := sets.NewString(values...)
 
-	for _, value := range values {
-		if temp.Has(value) {
-			repeated.Insert(value)
-		} else {
-			temp.Insert(value)
+		if count == 0 {
+			ret = valuesSet
+			count++
+			continue
 		}
+
+		ret = ret.Intersection(valuesSet)
+		count++
 	}
 
-	return repeated
+	return ret
```

Comment:
This can be IMO simplified as:

```
	var common sets.String
	for _, v := range privateReposWithValues {
		vs := sets.NewString(v...)
		if common == nil {
		 	common = vs
		} else {
			common = common.Intersection(vs)
		}
	}
	
	return common
```

Reply:
I had it like this but the problem with this approach is that if the intersection becomes nil in the middle of the iteration we will get the current values instead. That's why I used the counter instead.

Reply:
How can the intersection become nil in the middle? Intersection seems to returns an empty set (=empty map), not nil when the intersection is empty....

Reply:
I was curious so I tested, and it seems to behave correctly: https://play.golang.org/p/c0QmcLhxKk4

Reply:
I will address this. The nil check will just do the trick
Thanks.